src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -77,102 +69,55 @@ const BridgeConfirmation: FC = () => {
-                const newMaxPossibleAmountConsideringFee =
-                  isTokenEther && remainder && isRemainderPositive ? amount.sub(remainder) : amount;
-
-                const msTimeout = FADE_DURATION_IN_SECONDS * 1000;
-                const etherToken = isTokenEther ? token : getEtherToken(from);
-
-                const hasOnScreenFeeChanged =
-                  oldFee !== undefined &&
-                  formatTokenAmount(oldFee, etherToken) !== formatTokenAmount(newFee, etherToken);
-
-                setShouldUpdateOnScreenFee(hasOnScreenFeeChanged);
-
-                setMaxPossibleAmountConsideringFee((oldMaxPossibleAmountConsideringFee) => {
-                  const formattedOldMaxPossibleAmountConsideringFee =
-                    oldMaxPossibleAmountConsideringFee &&
-                    (oldMaxPossibleAmountConsideringFee.isNegative()
-                      ? "0"
-                      : formatTokenAmount(oldMaxPossibleAmountConsideringFee, token));
-
-                  const formattedNewMaxPossibleAmountConsideringFee =
-                    newMaxPossibleAmountConsideringFee.isNegative()
-                      ? "0"
-                      : formatTokenAmount(newMaxPossibleAmountConsideringFee, token);
-
-                  const hasOnScreenAmountChanged =
-                    formattedOldMaxPossibleAmountConsideringFee !==
-                    formattedNewMaxPossibleAmountConsideringFee;
-
-                  setShouldUpdateOnScreenAmount(hasOnScreenAmountChanged);
-
-                  const areAmountsEqual =
-                    oldMaxPossibleAmountConsideringFee !== undefined &&
-                    oldMaxPossibleAmountConsideringFee.eq(newMaxPossibleAmountConsideringFee);
-
-                  if (!areAmountsEqual) {
-                    setTimeout(() => {
-                      setMaxPossibleAmountConsideringFee(newMaxPossibleAmountConsideringFee);
-                    }, msTimeout);
-                  }
-                  return oldMaxPossibleAmountConsideringFee;
-                });
+            const isTokenEther = token.address === ethersConstants.AddressZero;
 
-                setTimeout(() => {
-                  setEstimatedGas({
-                    status: "successful",
-                    data: newGas,
-                  });
-                  setIsFadeVisible(true);
-                }, msTimeout);
+            if (isTokenEther) {
+              const amountIncludingFee = amount.add(newFee);
+              const tokenBalanceRemainder = amountIncludingFee.sub(tokenBalance);
+              const doesAmountExceedsTokenBalance = tokenBalanceRemainder.isNegative();
+              const newMaxAmountIncludingFee =
+                isTokenEther && !doesAmountExceedsTokenBalance
+                  ? amount.sub(tokenBalanceRemainder)
+                  : amount;
+
+              setMaxAmountIncludingFee(newMaxAmountIncludingFee);
+            } else {
+              setMaxAmountIncludingFee(amount);
+            }
```

Comment:
The block that updates the amount should be inside the `setTimeout` and both the fiat and the token amount should have the fade class: 

```diff
diff --git a/src/views/bridge-confirmation/bridge-confirmation.view.tsx b/src/views/bridge-confirmation/bridge-confirmation.view.tsx
index 7a1aea7..0355e9d 100644
--- a/src/views/bridge-confirmation/bridge-confirmation.view.tsx
+++ b/src/views/bridge-confirmation/bridge-confirmation.view.tsx
@@ -99,23 +99,21 @@ const BridgeConfirmation: FC = () => {

             const isTokenEther = token.address === ethersConstants.AddressZero;

-            if (isTokenEther) {
-              const amountIncludingFee = amount.add(newFee);
-              const tokenBalanceRemainder = amountIncludingFee.sub(tokenBalance);
-              const doesAmountExceedsTokenBalance = tokenBalanceRemainder.isNegative();
-              const newMaxAmountIncludingFee =
-                isTokenEther && !doesAmountExceedsTokenBalance
-                  ? amount.sub(tokenBalanceRemainder)
-                  : amount;
-
-              setMaxAmountIncludingFee(newMaxAmountIncludingFee);
-            } else {
-              setMaxAmountIncludingFee(amount);
-            }
-
             setFadeClass(classes.fadeOut);
             setTimeout(() => {
               setEstimatedGas({ status: "successful", data: gas });
+              if (isTokenEther) {
+                const amountIncludingFee = amount.add(newFee);
+                const tokenBalanceRemainder = amountIncludingFee.sub(tokenBalance);
+                const doesAmountExceedsTokenBalance = tokenBalanceRemainder.isNegative();
+                const newMaxAmountIncludingFee = !doesAmountExceedsTokenBalance
+                  ? amount.sub(tokenBalanceRemainder)
+                  : amount;
+
+                setMaxAmountIncludingFee(newMaxAmountIncludingFee);
+              } else {
+                setMaxAmountIncludingFee(amount);
+              }
               setFadeClass(classes.fadeIn);
             }, FADE_DURATION_IN_MS);
           })
@@ -446,9 +444,11 @@ const BridgeConfirmation: FC = () => {
       <Header title="Confirm Bridge" backTo={{ routeKey: "home" }} />
       <Card className={classes.card}>
         <Icon url={token.logoURI} size={46} className={classes.tokenIcon} />
-        <Typography type="h1">{tokenAmountString}</Typography>
+        <Typography type="h1" className={fadeClass}>
+          {tokenAmountString}
+        </Typography>
         {fiatAmountString && (
-          <Typography className={classes.fiat} type="body2">
+          <Typography className={`${classes.fiat} ${fadeClass || ""}`} type="body2">
             {fiatAmountString}
           </Typography>
         )}
```

Reply:
Yep, i've added it back again

Reply:
Comparing the BigNumbers will result in unnecessary fades due to the display precision.

Reply:
This also breaks with negative amounts:

![negative-values-fade](https://user-images.githubusercontent.com/15896177/197736094-c839290d-38e6-4928-a281-969531c9e758.gif)

All this was there for a reason:

![image](https://user-images.githubusercontent.com/15896177/197736294-ce4e7173-5771-422f-9d09-14a367cc9bf0.png)